cmd/ebrelayer/txs/relayToEthereum.go

```diff
@@ -20,64 +23,87 @@ import (
 const (
 	// GasLimit the gas limit in Gwei used for transactions sent with TransactOpts
 	GasLimit            = uint64(500000)
-	transactionInterval = 60 * time.Second
 )
 
-// RelayProphecyClaimToEthereum relays the provided ProphecyClaim to CosmosBridge contract on the Ethereum network
-func RelayProphecyClaimToEthereum(provider string, contractAddress common.Address, event types.Event,
-	claim ProphecyClaim, key *ecdsa.PrivateKey, sugaredLogger *zap.SugaredLogger) error {
-	// Initialize client service, validator's tx auth, and target contract address
-	client, auth, target, err := initRelayConfig(provider, contractAddress, event, key, sugaredLogger)
-	if err != nil {
-		sugaredLogger.Errorw("failed in init relay config.",
-			errorMessageKey, err.Error())
-		return err
-	}
+var GasPriceMinimum *big.Int = big.NewInt(60000000000)
 
-	// Initialize CosmosBridge instance
-	cosmosBridgeInstance, err := cosmosbridge.NewCosmosBridge(target, client)
-	if err != nil {
-		sugaredLogger.Errorw("failed to get cosmosBridge instance.",
-			errorMessageKey, err.Error())
-		return err
-	}
+func sleepThread(seconds time.Duration) {
+	time.Sleep(time.Second * seconds)
+}
+
+// RelayProphecyClaimToEthereum relays the provided ProphecyClaim to CosmosBridge contract on the Ethereum network
+func RelayProphecyClaimToEthereum(
+	claim ProphecyClaim,
+	sugaredLogger *zap.SugaredLogger,
+	client *ethclient.Client,
+	auth *bind.TransactOpts,
+	cosmosBridgeInstance *cosmosbridge.CosmosBridge,
+	) error {
 
 	// Send transaction
-	sugaredLogger.Infow("Sending new ProphecyClaim to CosmosBridge.",
+	sugaredLogger.Infow(
+		"Sending new ProphecyClaim to CosmosBridge.",
 		"CosmosSender", claim.CosmosSender,
-		"CosmosSenderSequence", claim.CosmosSenderSequence)
-
-	tx, err := cosmosBridgeInstance.NewProphecyClaim(auth, uint8(claim.ClaimType),
-		claim.CosmosSender, claim.CosmosSenderSequence, claim.EthereumReceiver, claim.Symbol, claim.Amount.BigInt())
+		"CosmosSenderSequence", claim.CosmosSenderSequence,
+	)
+
+	tx, err := cosmosBridgeInstance.NewProphecyClaim(
+		auth, uint8(claim.ClaimType), claim.CosmosSender,
+		claim.CosmosSenderSequence, claim.EthereumReceiver,
+		claim.Symbol, claim.Amount.BigInt(),
+	)
+	
+	// sleep 2 seconds to wait for tx to go through.
+	sleepThread(2)
```

Comment:
could move the sleep after the error check

Reply:
I moved the sleep here before the error check on purpose. This way, if infura is down, we wait two seconds before re-broadcasting that same transaction so it gives them time to go back online.